src/components/shared/Entity/DetailsForm/useDataPlaneField.ts

```diff
@@ -0,0 +1,205 @@
+
+                            return;
+                        }
+
+                        if (response.data) {
+                            const formattedData = response.data.map(
+                                ({ data_plane_name, id }) => {
+                                    const scope =
+                                        getDataPlaneScope(data_plane_name);
+
+                                    const {
+                                        cluster,
+                                        prefix,
+                                        provider,
+                                        region,
+                                    } = parseDataPlaneName(
+                                        data_plane_name,
+                                        scope
+                                    );
+
+                                    return {
+                                        dataPlaneName: {
+                                            cluster,
+                                            prefix,
+                                            provider,
+                                            region,
+                                            whole: data_plane_name,
+                                        },
+                                        id,
+                                        scope,
+                                    };
+                                }
+                            );
+
+                            setOptions(formattedData);
+                        }
+                    },
+                    (error) => {
+                        setServerError(error);
+                    }
+                )
+                .finally(() => {
+                    setLoading(false);
+                });
+        }
+    }, [dataPlaneOption, setLoading, setOptions]);
+
+    const dataPlaneSchema = useMemo(() => {
+        const dataPlanesOneOf: OneOfElement[] = [];
+
+        if (options.length > 0) {
+            options.forEach((option) => {
+                const {
+                    cluster,
+                    provider,
+                    region,
+                    whole: wholeName,
+                } = option.dataPlaneName;
+
+                const title = formatDataPlaneName(
+                    cluster,
+                    provider,
+                    region,
+                    wholeName
+                );
+
+                dataPlanesOneOf.push({ const: option, title });
+            });
+        }
+
+        return dataPlaneOption === 'show_option' && hasLength(dataPlanesOneOf)
+            ? {
+                  [DATA_PLANE_SCOPE]: {
+                      description: intl.formatMessage({
+                          id: 'workflows.dataPlane.description',
+                      }),
+                      oneOf: dataPlanesOneOf,
+                      type: 'object',
+                  },
+              }
+            : null;
```

Comment:
If we are going to check 

```
dataPlaneOption === 'show_option'
```

at the end we should just check that before running anything and return null right at the start.